html_css/implementing_accessibility.md

```diff
@@ -0,0 +1,221 @@
+
+If you were to use only `<div>` elements to act as regions or headings (the elements `<h1>` through `<h6>`), then a screen reader user might have to go through the entire page just to get to a specific area. By properly using landmarks and headings, you provide users of assistive technologies a more operable page with more context. Not only can screen readers navigate a page via landmarks and headings by opening a menu in their screen reader, but these elements also have their roles announced. Watch [Why headings and landmarks are so important](https://www.youtube.com/watch?v=vAAzdi1xuUY&list=PLNYkxOF6rcICWx0C9LVWWVqvHlYJyqw7g&index=19) to see how a screen reader interacts with these elements, and to hear some words of caution when using headings.
+
+### Using Color
+
+Even though adding color to a page can make it more visually appealing, using the wrong color combination or relying solely on color to convey information can end up making your page more difficult to perceive for some users.
+
+#### Contrast Ratio
+
+![Contrast ratio issues](https://user-images.githubusercontent.com/70952936/125673709-dd42bdf0-a4bc-4103-9a1b-e73e6c3bc85b.jpg)
+
+Can you easily read the text in all three (yes, three) examples from the image above? If you can't, then that's because the contrast ratio is too low. If you can, then consider me jealous of your eye sight, but you still need to provide an easily perceivable website for visually impaired users.
+
+Contrast ratio refers to the color of foreground text against the color of a background, and includes images of text. There are two different WCAG conformity levels for contrast ratios, both of which have rules for normal text and large text. **Normal text** is defined as text with a font size that's less than 18 points/24px (or less than 14 points/18.66px for bold text), and **large text** is defined as text with a font size that is at least 18 points/24px (or at least 14 points/18.66px for bold text).
+
+1. **Level AA** (minimum) requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text.
+2. **Level AAA** (enhanced) requires a contrast ration of at least 7:1 for normal text and 4.5:1 for large text.
+
+Both levels have exceptions that don't need to adhere to the required contrast ratio rule:
+
+* Incidental text, such as text that just happens to be within an image that has other significant visual content, or text that is purely decorative.
+* Text that is part of an inactive or disabled user interface component, such as a button that is disabled and has a lowered opacity.
+* Text that is part of a logo or brand name.
+
+[WebAIM Contrast Checker](https://webaim.org/resources/contrastchecker/) is a fantastic tool for checking the contrast ratio. The page also has a link for a link contrast checker, which goes over what the contrast ratio should be if a text link isn't underlined (its default style).
```

Comment:
So my reasoning for not mentioning it here was more because I wanted to lump it in with the auditing lesson, since that goes into using the dev tools for checking stuff like this; sort of "here's a tool you can use" in this lesson and then follow up in a later lesson with "hey here's a better/alternative way", and also keep the dev tools stuff together. But I can see your point here. 

If you think quickly mentioning how to use the "element picker" tool in Chrome's dev tools to check contrasts here would make sense I could add that. My concern would kind of be twofold: either leaving Firefox out, or mentioning how to use the Firefox contrast tools as well and having to go over the same steps that are mentioned in the Auditing lesson (not that they're complicated steps or anything).

Reply:
No need to go into deep detail about it, but it's worth mentioning here for completeness sake. 

a. We'll get people telling us that it exists in the chat a lot if we don't
b. It's worth letting them do a bit of digging themselves to identify other tools they may not know about. It's OK to explicitly explain it later too.

Reply:
Makes sense. Maybe something along the lines of:

"You can also check the contrast ratio in your browser's dev tools. In Chrome, you would click the "element picker" tool in the Elements tab, then hover over an element on the web page. In Firefox, you can do the same thing, except you have to click the "accessible object picker" tool in the Accessibility tab instead."

And of course updating some verbiage in the Auditing lesson.